src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Configuration/BaseProjectConfigurationDimensionProvider.cs

```diff
@@ -130,5 +139,69 @@ public BaseProjectConfigurationDimensionProvider(IProjectXmlAccessor projectXmlA
         {
             return await _projectXmlAccessor.GetEvaluatedPropertyValue(unconfiguredProject, _propertyName).ConfigureAwait(false);
         }
+
+        /// <summary>
+        /// Adds a dimension value to the project.
+        /// </summary>
+        /// <param name="unconfiguredProject">Unconfigured project for which the configuration change.</param>
+        /// <param name="dimensionValue">Name of the new dimension value.</param>
+        /// <returns>A task for the async operation.</returns>
+        protected async Task OnDimensionValueAddedAsync(UnconfiguredProject unconfiguredProject, string dimensionValue)
+        {
+            string evaluatedPropertyValue = await GetPropertyValue(unconfiguredProject).ConfigureAwait(false);
+            await _projectXmlAccessor.ExecuteInWriteLock(msbuildProject =>
+            {
+                MsBuildUtilities.AppendPropertyValue(msbuildProject, evaluatedPropertyValue, _propertyName, dimensionValue);
+            }).ConfigureAwait(false);
+
+            _telemetryService.PostProperty($"DimensionChanged/{_dimensionName}/Add", "Value", HashValueIfNeeded(dimensionValue), unconfiguredProject);
```

Comment:
these constant seems used multiple times, how about having const or event name getter to reduce typo issue in future?